metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -265,10 +304,71 @@ def _extract_record(self, topic: str) -> Iterable[MetadataWorkUnit]:
                 self.report.report_workunit(wu)
                 yield wu
 
+    def build_custom_properties(self, topic_detail, extra_topic_config):
+
+        replication_factor = None
+        for _, p_meta in topic_detail.partitions.items():
+            if replication_factor is None or len(p_meta.replicas) > replication_factor:
+                replication_factor = len(p_meta.replicas)
+
+        custom_props = {
+            "Partitions": len(topic_detail.partitions),
+            "Replication Factor": replication_factor,
+        }
+        if extra_topic_config is not None:
+            for config_key in KafkaTopicConfigKeys:
+                custom_props[config_key] = self._get_config_value_if_present(
+                    extra_topic_config, config_key
+                )
+
+        return {
+            k: (json.dumps(v) if not isinstance(v, str) else v)
```

Comment:
Any chance the json dumps fails? Do we need to catch this case?

Reply:
updated the handling.